src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -916,13 +926,50 @@
     ;; default to itself
     mount))
 
+(defn read-bolt-project-config
+  [project-dir]
+  (let [config-path (str project-dir "/bolt-project.yaml")]
+    (if (fs/file? config-path)
+     (yaml/parse-string (slurp config-path)))))
+
+(def default-project-modulepath
+  ["modules" "site-modules" "site"])
+
+(defn parse-modulepath
+  "The modulepath for a bolt project can either be an array of paths or a string
+  with a path separator."
+  [modulepath]
+  (if (string? modulepath)
+    (str/split modulepath #":")
+    modulepath))
+
+(defn get-project-modulepath
+  "Given a map representing the bolt project configuration, return the
+  modulepath. If the modulepath is not defined or the configuration map passed
+  is nil then return the default modulepath."
+  [project-config]
+  (let [has-modules-key? (contains? project-config :modules)]
+    (if-let [modulepath (parse-modulepath (get project-config :modulepath))]
+      (if has-modules-key?
+        (concat modulepath [".modules"])
+        modulepath)
+      (if has-modules-key?
+        ["modules"]
+        default-project-modulepath))))
+
 (defn find-project-file
   "Find a file in a project using the parameters from a `file_content` request.
-  Returns the path as as string."
-  [bolt-projects-dir project-name mount module path]
-  (if (is-bolt-project? (str bolt-projects-dir "/" project-name))
-    (let [project-root (get-project-root bolt-projects-dir project-name)
-          module-root (find-project-module project-root module)
+  Returns the path as as string. If the module name is the same as the project
+  name then files are served from the project directly."
+  [bolt-projects-dir project-ref mount module path]
+  (if (is-bolt-project? (str bolt-projects-dir "/" project-ref))
+    (let [project-root (get-project-root bolt-projects-dir project-ref)
+          project-config (read-bolt-project-config project-root)
+          project-name (get project-config :name)
```

Comment:
Is this the only way a project's name is defined? What if `project-config` is nil?

Reply:
Bolt documents that in order to load top level files as a module then you have to define this name explicitly in the project config. [https://github.com/puppetlabs/bolt/issues/1871]

If project-config is nil then project-name will be nil.

Reply:
okay, I was wondering if we needed to fall back to the name from `project-ref` or something, but if Bolt requires a `:name` to do load top-level files, then sounds like we're good. 👍🏼